src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -66,14 +66,33 @@ export const NavSearchBar: React.FC<NavSearchBarProps> = ({ venueId }) => {
     currentVenueId: venueId,
   });
 
-  const relatedRooms = useMemo<Room[]>(
+  const enabledRelatedRooms = useMemo<Room[]>(
     () =>
       relatedVenues
         .flatMap((venue) => venue.rooms ?? [])
-        .filter((room) => room),
+        .filter((room) => {
+          if (isDefined(room.type) && COVERT_ROOM_TYPES.includes(room.type)) {
+            return false;
+          }
+
+          return room.isEnabled;
+        }),
     [relatedVenues]
   );
 
+  const enabledRelatedRoomsByTitle = useMemo<Partial<Record<string, Room>>>(
+    () =>
+      reduce(
+        enabledRelatedRooms,
+        (enabledRelatedRoomsByTitle, room) => ({
+          ...enabledRelatedRoomsByTitle,
+          [room.title]: room,
+        }),
+        {}
+      ),
+    [enabledRelatedRooms]
+  );
```

Comment:
There's no need to use `reduce` from `lodash`. The language already has built in functionality for this: `theArray.reduce`. This needs to be reverted and fixed in a followup PR.